libexec/pyenv-rehash

```diff
@@ -32,6 +66,9 @@ trap remove_prototype_shim EXIT
 
 remove_prototype_shim() {
   rm -f "$PROTOTYPE_SHIM_PATH"
+  if [ -n "$PYENV_REHASH_PARALLEL_SAFE" ]; then
+    release_lock
```

Comment:
The release of lock need to be done as a `EXIT` trap (e.g. `trap release_lock EXIT`) to make sure the lock is released at the end of the script.

Reply:
This function is already registered as an exit trap, but I've renamed it so hopefully its purpose is clearer.